src/image.rs

```diff
@@ -423,16 +428,36 @@ pub trait ImageDecoder<'a>: Sized {
     }
 
     /// Returns all the bytes in the image.
-    fn read_image(self) -> ImageResult<Vec<u8>> {
+    ///
+    /// This function takes a slices of bytes and writes the pixel data of the image into it.
+    /// Although not required, for certain color types callers may want to pass buffers which are
+    /// aligned to 2 or 4 byte boundries to the slice can be cast to a [u16] or [u32]. To accommodate
+    /// such casts, the returned contents will always be in native endian.
+    ///
+    /// # Panics
+    ///
+    /// This function panics if buf.len() != self.total_bytes().
+    ///
+    /// # Examples
+    ///
+    /// ```no_run
+    /// use zerocopy::{AsBytes, FromBytes};
```

Comment:
This idea to add `zerocopy` as a new dependency suggests itself basically. If so, that seems interesting to consider. It could make the implementation itself safe although I would want to avoid adding it to the public interface if possible.